###author, forum, title, text, 여태까지 한 것들을 불러와서 하나로 합침

In [ ]:
import pandas as pd
import numpy as np
import json
import pickle

max_features = 1000
author_forum_p = pickle.load(open("author, forum_%d.p" % max_features, "rb"))
title_p = pickle.load(open("title_%d.p" % max_features, "rb"))
text_p = pickle.load(open("text_%d.p" % max_features, "rb"))

train = pd.concat([author_forum_p, title_p, text_p], axis = 1)

train

### predictor, model 생성

In [14]:
from sklearn.ensemble import RandomForestClassifier

label = 'istroll'
pre = train.columns.drop([label])

model = RandomForestClassifier(n_estimators=10, n_jobs=6)

###cross validation

In [15]:
from sklearn import cross_validation

cv_value = 3

scores = cross_validation.cross_val_score(model, train[pre], train[label], cv=cv_value, scoring="roc_auc")
cv_result = scores.mean()

print(cv_result)

0.72148840098


##Result
   * forumid와 author, title, text를 bag of words로 변환 (feature : 1,000개) - 0.721
   * forumid와 author, title을 bag of words로 변환 (feature : 1,000개) - 0.712577335639
   * forumid와 author, title을 bag of words로 변환 (feature : 1,000개) - 0.712577335639